File: momentra/momentra/Shell/Personal/Shared/PersonalMasterExpenseIcons.swift
import SwiftUI

/// Native SF Symbols for Master Expense UI chrome (categories + structural UI).
enum PersonalMasterExpenseIcons {
    static func categorySymbol(code: String) -> String {
        switch code.uppercased() {
        case "FOOD": return "fork.knife"
        case "TRANSPORT": return "car.fill"
        case "SHOPPING": return "bag.fill"
        case "CAFE": return "cup.and.saucer.fill"
        case "HEALTH": return "cross.case.fill"
        case "ENTERTAINMENT": return "film.fill"
        case "BILLS": return "doc.text.fill"
        default: return "shippingbox.fill"
        }
    }

    enum Chrome: String {
        case header = "creditcard.fill"
        case info = "info.circle.fill"
        case edit = "pencil"
        case amount = "indianrupeesign.circle.fill"
        case calendar = "calendar"
        case folder = "folder.fill"
        case back = "chevron.left"
        case expandUp = "chevron.up"
        case expandDown = "chevron.down"
    }
}

struct MeIcon: View {
    let symbol: String
    var tint: Color = .primary
    var size: CGFloat = 20
    var accessibilityLabel: String?

    var body: some View {
        Image(systemName: symbol)
            .font(.system(size: size))
            .foregroundStyle(tint)
            .accessibilityLabel(accessibilityLabel ?? "")
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/personal/shared/PersonalMasterExpenseIcons.kt
package com.example.momentra.ui.shell.personal.shared

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowBack
import androidx.compose.material.icons.outlined.CalendarToday
import androidx.compose.material.icons.outlined.CreditCard
import androidx.compose.material.icons.outlined.CurrencyRupee
import androidx.compose.material.icons.outlined.Edit
import androidx.compose.material.icons.outlined.Folder
import androidx.compose.material.icons.outlined.Info
import androidx.compose.material.icons.outlined.KeyboardArrowDown
import androidx.compose.material.icons.outlined.KeyboardArrowUp
import androidx.compose.material.icons.outlined.LocalCafe
import androidx.compose.material.icons.outlined.Movie
import androidx.compose.material.icons.outlined.ReceiptLong
import androidx.compose.material.icons.outlined.Restaurant
import androidx.compose.material.icons.outlined.ShoppingBag
import androidx.compose.material.icons.outlined.DirectionsCar
import androidx.compose.material.icons.outlined.Inventory2
import androidx.compose.material.icons.outlined.MedicalServices
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.layout.size

/** Native Material Outlined icons for Master Expense UI chrome (categories + structural UI). */
object PersonalMasterExpenseIcons {
    fun categoryIcon(code: String): ImageVector = when (code.uppercase()) {
        "FOOD" -> Icons.Outlined.Restaurant
        "TRANSPORT" -> Icons.Outlined.DirectionsCar
        "SHOPPING" -> Icons.Outlined.ShoppingBag
        "CAFE" -> Icons.Outlined.LocalCafe
        "HEALTH" -> Icons.Outlined.MedicalServices
        "ENTERTAINMENT" -> Icons.Outlined.Movie
        "BILLS" -> Icons.Outlined.ReceiptLong
        else -> Icons.Outlined.Inventory2
    }

    enum class Chrome(val vector: ImageVector) {
        Header(Icons.Outlined.CreditCard),
        Info(Icons.Outlined.Info),
        Edit(Icons.Outlined.Edit),
        Amount(Icons.Outlined.CurrencyRupee),
        Calendar(Icons.Outlined.CalendarToday),
        Folder(Icons.Outlined.Folder),
        Back(Icons.AutoMirrored.Outlined.ArrowBack),
        ExpandUp(Icons.Outlined.KeyboardArrowUp),
        ExpandDown(Icons.Outlined.KeyboardArrowDown),
    }
}

@Composable
fun MeIcon(
    icon: ImageVector,
    contentDescription: String?,
    modifier: Modifier = Modifier,
    tint: Color = Color.Unspecified,
    size: Dp = 20.dp,
) {
    Icon(
        imageVector = icon,
        contentDescription = contentDescription,
        modifier = modifier.size(size),
        tint = tint,
    )
}
